allennlp/modules/seq2seq_encoders/pytorch_seq2seq_wrapper.py

```diff
@@ -29,76 +31,155 @@ class PytorchSeq2SeqWrapper(Seq2SeqEncoder):
     Note that we *require* you to pass sequence lengths when you call this module, to avoid subtle
     bugs around masking.  If you already have a ``PackedSequence`` you can pass ``None`` as the
     second parameter.
+
+    We support stateful RNNs where the final state from each batch is used as the initial
+    state for the subsequent batch by passing ``stateful=True`` to the constructor.  In this case,
+    ``max_batch_size`` is the maximum batch size allowed (although batches of any size less
+    then the maximum are also supported).  If ``stateful=False`` then ``max_batch_size``
+    is ignored.
+
+    We also support stacked RNNs that return activations for each layer by passing ``stacked=True``
+    to the constructor.  In this case, the ``module`` forward method has a slightly different
+    signature from ``torch.nn.modules.RNNBase``.  It returns:
+
+        - hidden states of size ``(num_layers, batch_size, timesteps, hidden_dim)``
+        - final states, a tuple of sizes ``(num_layers, batch_size, hidden_dim)``
+          and ``(num_layers, batch_size, memory_dim)``
+
     """
-    def __init__(self, module: torch.nn.modules.RNNBase) -> None:
```

Comment:
The reason this was an `RNNBase` was because they all define `input_size`, `hidden_size`. If we are going to switch to allowing `Modules` we should make it very clear in the docstring (or even raise errors) when they do not define these attributes.

Reply:
It's already clear in the existing docstring that the module must have this interface.  If someone implements a module that doesn't have this interface then an exception will be raised when these attributes are accessed.